packages/arb-token-bridge-ui/src/components/TransferPanel/DepositCardL2Failure.tsx

```diff
@@ -19,40 +30,75 @@ export function DepositCardL2Failure({ tx }: { tx: MergedTransaction }) {
     [isConnectedToArbitrum]
   )
 
+  useEffect(() => {
+    ;(async () => {
+      const { daysTillExpiry, isValid } = await getRetryableTicketExpiration({
+        l1TxHash: tx.txId,
+        l1Provider: l1Signer.provider,
+        l2Provider: l2Signer.provider
+      })
+
+      setRetryableExpiryDays({
+        days: daysTillExpiry,
+        isValid
+      })
+    })()
+  }, [l1Signer.provider, l2Signer.provider, tx.txId])
+
   return (
-    <DepositCardContainer tx={tx}>
-      <span className="text-4xl font-semibold text-orange-dark">
-        {isRedeeming ? 'Re-executing...' : 'L2 transaction failed'}
-      </span>
-      <Tooltip
-        show={isRedeemButtonDisabled}
-        content={
-          <span>
-            Please connect to the L2 network to re-execute your deposit.
+    <>
+      {/* Only show the CTA and `days remaining to retry..` modal if the remaining days are valid */}
+      {retryableExpiryDays?.isValid && (
+        <DepositCardContainer tx={tx}>
+          <span className="text-4xl font-semibold text-orange-dark">
+            {isRedeeming ? 'Re-executing...' : 'L2 transaction failed'}
+          </span>
+
+          <div className="h-1" />
+
+          <span className="text-2xl font-normal text-orange-dark">
+            No worries, we can try again. You have{' '}
+            {retryableExpiryDays.days > 0
+              ? retryableExpiryDays.days > 1
+                ? `${retryableExpiryDays.days} days`
+                : `${retryableExpiryDays.days} day`
+              : 'less than a day'}{' '}
```

Comment:
Let's extract this into a `getRetryText` function and get rid of the nested ternaries.

Reply:
Extracted it out to a memoized value `retryableExpiryDaysText` dependent on days remaining, so that the function doesn't get unnecessarily called again on each re-render.